languages/multi-language/src/main/java/de/jplag/multilang/MultiLanguageParser.java

```diff
@@ -7,18 +7,28 @@
 import java.util.Optional;
 import java.util.Set;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import de.jplag.Language;
 import de.jplag.ParsingException;
 import de.jplag.Token;
 
 public class MultiLanguageParser {
+    private static final Logger LOG = LoggerFactory.getLogger(MultiLanguageParser.class);
+    private static final String WARNING = "This module only allows parsing of multiple languages. No comparisons will be made between languages";
+    private static final String ERROR_MULTIPLE_PRIORITY_LANGUAGES = "Multiple language modules with priority (%s) have been found for file: %s";
+    private static final String ERROR_MULTIPLE_LANGUAGES = "Multiple language modules (%s) have been found for file: %s";
     private final List<Language> languages;
+    private static boolean hasPrintedWarning;
 
     public MultiLanguageParser(MultiLanguageOptions options) {
         this.languages = options.getLanguages();
+        hasPrintedWarning = false;
     }
 
     public List<Token> parseFiles(Set<File> files, boolean normalize) throws ParsingException {
+        this.printWarning();
         List<Token> results = new ArrayList<>();
         for (File file : files) {
             Optional<Language> language = findLanguageForFile(file);
```

Comment:
We discussed that we can find out at compile time whether the mapping of file extensions to language modules is well-defined. I would argue then that we should set up an actual look up table instead of repeating the same steps for each individual file.

Reply:
Yes, I meant it as general information . I would not implement it in this PR :)

Reply:
@TwoOfTwelve Sorry for the unclear positioning of my comment. I was not referencing the warning, but the call to `findLanguageForFile(file)`, which repeats the same tedious steps for each file. I propose that, instead, a `Map<String, Language>` of each file extension to the respective language module is set up during initialization, so that this "number of language modules" x "number of files" effort is reduced to a simple lookup of the file extension. The check for ambiguous mapping can also be done there one-time.